pkg/apis/workflow/v1alpha1/anystring.go

```diff
@@ -50,3 +51,15 @@ func (i AnyString) MarshalJSON() ([]byte, error) {
 func (i AnyString) String() string {
 	return string(i)
 }
+
+func (i AnyString) Value() interface{} {
```

Comment:
this is wrong, `AnyString` is only intended to convert other things to string, it should not return non-string values.

Reply:
parameters are ALWAYS string, but we should add functions to cast them to int/float64, I think we can use Sprig for that, see here how I do that:

https://github.com/argoproj/argo-workflows/pull/5115/files#diff-f2728b47cf70dd622efce9723f26cbdfb848f7b422a124732c3796aea84fbad5R134

Reply:
Agreed,  here below E.g.:
            valueFrom:
              expression: "inputs.parameters.num == '1'? '1': inputs.parameters.num == '2'? '1': steps.fibonaccihelper.outputs.parameters.result"